v2/tools/generator/internal/astmodel/property_definition.go

```diff
@@ -156,6 +158,17 @@ func (property *PropertyDefinition) SetFlatten(flatten bool) *PropertyDefinition
 	return &result
 }
 
+// SetSecret sets if the property is a secret
+func (property *PropertyDefinition) SetSecret(secret bool) *PropertyDefinition {
+	if secret == property.secret {
+		return property
+	}
+
+	result := *property
```

Comment:
We have a helper for clarity:

```suggestion
	result := property.copy()
```